### ISSUE TITLE
Node stopped with multiple PoST services

### ISSUE THREAD

**Issue by Hugh-Top** [spacemeshos/go-spacemesh#5752]
## Description

node:post service = 1:N
node alway stopped by panic

## Steps to reproduce
1. nohup go-spacemesh --config /data/smh/config.mainnet.json --listen /ip4/0.0.0.0/tcp/7513 --smeshing-opts-provider 4294967295 -d /data/smh/data/ >> /data/logf/spacemesh.log 2>&1 &
2.  Start part of the post service

## Actual Behavior

```shell
2024-03-21T09:49:15.315Z        INFO    node.post       post state changed      {"id": "18ef72255dxxxxxf2d08a2739828070ae4a13c7ad0b4", "state": "idle"}
2024-03-21T09:49:15.315Z        INFO    node.atxBuilder atx builder stopped
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x1235df2]

goroutine 38024 [running]:
github.com/spacemeshos/go-spacemesh/activation.(*Builder).buildInitialPost(0xc0b9664000, {0x1eb1f88, 0xc0b8fd5220}, {0x18, 0xef, 0x72, 0x25, 0x5d, 0x75, 0x44, ...})
        /data/dev/go-spacemesh/activation/activation.go:335 +0x2d2
github.com/spacemeshos/go-spacemesh/activation.(*Builder).run(0xc0b9664000, {0x1eb1f88, 0xc0b8fd5220}, 0xc0002f8500)
        /data/dev/go-spacemesh/activation/activation.go:360 +0x1cb
github.com/spacemeshos/go-spacemesh/activation.(*Builder).startID.func1()
        /data/dev/go-spacemesh/activation/activation.go:237 +0x25
golang.org/x/sync/errgroup.(*Group).Go.func1()
        /home/fsw/go/pkg/mod/golang.org/x/sync@v0.6.0/errgroup/errgroup.go:78 +0x56
created by golang.org/x/sync/errgroup.(*Group).Go in goroutine 1
        /home/fsw/go/pkg/mod/golang.org/x/sync@v0.6.0/errgroup/errgroup.go:75 +0x96
```

## Environment
Please complete the following information:
 - OS:  Ubuntu 22.04.3 LTS
 
 - Node Version: 1.4.2
 
 - Smapp version (if applicable):

**Comment by fasmat** [comment]
Hi @Hugh-Top 

can you check in the `postdata_metadata.json` of your PoST data dirs if any of those is missing the `Nonce` value in that file?

**Comment by fasmat** [comment]
Most likely your PoST data hasn't finished initializing and therefor cannot be used to create the initial proof for that identity (yet).

**Comment by Hugh-Top** [comment]
tks, as you said，i try

**Comment by Hugh-Top** [comment]
@fasmat  tks， node is normal after deleted PoST data without Nonce.
Why is Nonce not generated?
 Is there any way to avoid it? 
 It wastes a long time to generate data if there is no nonce

**Comment by fasmat** [comment]
You have to fully initialize the PoST data before you can use a post service with it. If the data is incomplete you will not be able to generate proofs with the PoST service. You can verify the PoST data with `postcli` (https://github.com/spacemeshos/post/releases/tag/v0.12.4) it can continue initialization if the data is incomplete and can also be used to look for a missing nonce.

**Comment by fasmat** [comment]
Fixed in #5753
